Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -441,47 +455,103 @@ public async Task ProcessExecutionResult(SubmissionExecutionResult submissionExe
         submission.Processed = true;
         submission.ProcessingComment = null;
 
+        using var scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
         if (executionResult != null)
         {
             ProcessTestsExecutionResult(submission, executionResult);
+
+            await this.submissionsForProcessingData.MarkProcessed(submission.Id);
             this.submissionsData.Update(submission);
             await this.submissionsData.SaveChanges();
```

Comment:
I think the MarkProcessed contains SaveChanges which can potentially lead to a case where the submissionForProcessing is marked as processed, but the submission result is not presenting in the database

Reply:
This code is in a transaction scope so even if UpdateResults throws an exception, nothing will be saved in the db no matter how many times we call the SaveChanges method.